src/route/seg6.rs

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: MIT
+
+use anyhow::Context;
+use byteorder::{ByteOrder, NativeEndian};
+use netlink_packet_utils::{
+    nla::{DefaultNla, Nla, NlaBuffer},
+    Parseable,
+};
+use std::net::Ipv6Addr;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+#[non_exhaustive]
+pub enum Seg6Mode {
+    // Inline mode for Seg6
+    Inline,
+    // Encapsulation mode for Seg6
+    Encap,
+    // L2ENCAP = 2,
+    // ENCAP_RED = 3,
+    // L2ENCAP_RED = 4
+    Other(u32),
+}
+
+impl From<Seg6Mode> for u32 {
+    fn from(value: Seg6Mode) -> Self {
+        match value {
+            Seg6Mode::Inline => 0,
+            Seg6Mode::Encap => 1,
```

Comment:
Instead of magic number, please create a constant for this using the same name kernel is using
```
SEG6_IPTUN_MODE_INLINE
SEG6_IPTUN_MODE_ENCAP
SEG6_IPTUN_MODE_L2ENCAP
SEG6_IPTUN_MODE_ENCAP_RED
SEG6_IPTUN_MODE_L2ENCAP_RED
```